Library/Homebrew/cmd/update.rb

```diff
@@ -263,18 +264,29 @@ def pull!(options = {})
 
     @current_revision = read_current_revision
 
-    if @initial_branch != "master" && !@initial_branch.empty?
+    if @initial_branch != @upstream_branch && !@initial_branch.empty?
       safe_system "git", "checkout", @initial_branch, *@quiet_args
+      pop_stash
+    else
+      pop_stash_message
     end
+  end
 
-    if @stashed
-      safe_system "git", "stash", "pop", *@quiet_args
-      if ARGV.verbose?
-        puts "Restored your changes:"
-        system "git", "status", "--short", "--untracked-files"
-      end
-      @stashed = false
+  def pop_stash
+    return unless @stashed
+    safe_system "git", "stash", "pop", *@quiet_args
+    if ARGV.verbose?
+      puts "Restoring your stashed changes to #{repository}:"
+      system "git", "status", "--short", "--untracked-files"
     end
+    @stashed = false
+  end
+
+  def pop_stash_message
+    return unless @stashed
+    puts "To restore the stashed changes to #{repository} run:"
+    puts "  `cd #{repository} && git stash pop`"
+    @stashed = false
```

Comment:
I understand why this is here, but I'm afraid that users won't notice this message (as it will be printed above the formula update report) and then they will complain that Homebrew destroyed their uncommitted local changes.

But let's keep it like this and if someone complains, we can tell them to `git stash pop`, and if too many people complain, we can still address this later.